### ISSUE TITLE
Incorrect value for field

### ISSUE THREAD

**Issue by EvgenyAfanasev** [AugustNagro/magnum#2]
Hi. I have next model:

```scala
  @Table(PostgresDbType, SqlNameMapper.CamelToSnakeCase)
  final case class Modules(
      @Id id: Int,
      title: String,
      parentId: Int,
      description: Option[String]
  )
  ```
  
  With next values:
  
  Modules(1, "Module title", 2, Some("Module Description"))
  
  And with next table definition:
  
 id INT
title VARCHAR(255)
description VARCHAR(255)
course_id INT 
  
  And when I tried add entity using `insertReturning` method I received error 
  
  `Incorrect value for int: Module Description`
  
I checked columns in my table definition and see problem with order of model attributes. I have to create ordering of case class properties like in database for working. Is it correct behavior ? IMHO, developer hasn't to think about ordering tables columns for creating dao layer

**Comment by AugustNagro** [comment]
We can change that by using the `ResultSet` columnLabel methods instead of columnIndex. Like [getInt(String)](https://docs.oracle.com/en/java/javase/20/docs/api/java.sql/java/sql/ResultSet.html#getInt(java.lang.String)) instead of [getInt(int)](https://docs.oracle.com/en/java/javase/20/docs/api/java.sql/java/sql/ResultSet.html#getInt(int))

Another option is to add an `@Order(int)` annotation and then put documentation in the readme.

I think I like Option 1 better. That's what spring data does and it's very user friendly, despite being less efficient (ResultSet javadoc says "In general, using the column index will be more efficient")

WDYT?

**Comment by EvgenyAfanasev** [comment]
Yeah, 1 option is more friendly

**Comment by AugustNagro** [comment]
Yep agreed. I think we will want to add `readSingle(rs: ResultSet, columnName: String): E` to DbCodec as well.

**Comment by EvgenyAfanasev** [comment]
But what do think about separation DbCodec on two parts: DbCodec as reader by column names and DbIndexedCodec as reader by index ? I can take it to work

**Comment by AugustNagro** [comment]
Sweet, thanks for taking it on. I am biased towards just having it all in
the one DbCodec, since there are no cases where you can do one but not the
other.

**Comment by lbialy** [comment]
@EvgenyAfanasev just a quick remark - tuples don't make much sense with named args but they do make a lot of sense with positional args. I guess derivation will have to make that choice and go with labels for case classes and indexes for tuples.

**Comment by AugustNagro** [comment]
Right. If we end up going with `readSingle(rs: ResultSet, columnName: String): E` I think it's ok to throw UnsupportedOperationException for Tuples. We should also add a check in the RepoDefaults macro to make sure each case class field's dbCodec.cols.length == 1, since nesting gets weird when using names, and it's a good practice for entity classes to be flat anyway.

**Comment by cornerman** [comment]
Just read this thread, and thought that the proposal for `named tuples` in scala might relevant here as well: https://github.com/scala/improvement-proposals/pull/72

I think, it would be great to support them when they are available.
